Add K3S Release Documentation (#6135)
k3s-io/k3s: docs/release/kubernetes-upgrade.md

## Clone and Setup Remotes
Clone from upstream then add k3s-io fork and your personal fork.
```sh
# initial clone
git clone --origin upstream \
  https://github.com/kubernetes/kubernetes.git \
  ${GOPATH}/src/github.com/kubernetes/kubernetes
 
cd ${GOPATH}/src/github.com/kubernetes/kubernetes
 
# add the k3s-io remote
git remote add k3s-io https://github.com/k3s-io/kubernetes.git
 
# fetch all remote branches and tags. If you receive a message saying that
# previous tags will be "clobbered", add --force to the command below.
git fetch --all --tags
```
# Rebasing and Generating Tags
Establish a local branch for rebasing:

```sh
export GLOBAL_GIT_CONFIG_PATH=$(git config --list --show-origin --show-scope --global | awk 'NR==1{ split($2,path,":"); print path[2] }')
export SSH_MOUNT_PATH=$(echo ${SSH_AUTH_SOCK} || echo "${HOME}/.ssh/id_rsa")

# Set up your new/old versions of Kubernetes
export OLD_K8S=<old-k8s-version>
export NEW_K8S=<new-k8s-version>
export OLD_K8S_CLIENT=<old-k8s-client-version>
export NEW_K8S_CLIENT=<new-k8s-client-version>

> export OLD_K3S_VER="{$OLD_K8S}-k3s1" 

Author: Good catch.

> export OLD_K3S_VER="${$OLD_K8S}-k3s1" 

Reviewer: double variable here

Reviewer: resolving for future iteration

Reviewer 2: This will be fixed before merge as this is an error. @nikolaishields

export NEW_K3S_VER="$NEW_K8S-k3s1"
export RELEASE_BRANCH=<k8s-release-branch>
export GOPATH=$(go env GOPATH)
 
# clean old builds
rm -rf _output
 
# Rebase k3s customizations from the previous release, minus the merge commit (~1),
# onto the new upstream tag. This will leave you on a detached head that will be
# tagged in the following step.
git rebase --onto $NEW_K8S $OLD_K8S $OLD_K3S_VER~1
 

> # Kubernetes is very picky about go versions. We use alpine and docker to specify the goversion with which we build the project.

Reviewer 2: ```suggestion
# Kubernetes is specific with the Go version used per release. We use alpine and docker to specify the Go version with which we build the project.
```

# Create a Release Candidate 
Releases are kicked off and created by tagging a new tag.
To create a new release in Github UI perform the following:

1. Set title and tag according to the release version you're working on. E.g. v1.22.5-rc1+k3s1.
2. Leave description blank.
3. Check the pre-release field.
4. Publish

The resulting run can be viewed here: 
[k3s-io/k3s Drone Dashboard](https://drone-publish.k3s.io/k3s-io/k3s)

# Create GA Release Candidate
Once QA has verified that the RC is good (or that any fixes have been added in follow up release candidates), it is time for the general release.

1. Create new release in the Github web interface
2. Set title: ${NEW_K8S}, add description with release notes. Leave the tag section blank.
3. Check the pre-release field.
4. Save as draft until RC testing is complete.

Once QA signs off on a RC:
1. Set tag to be created - this tag should match the tag in the drafted title.
2. Ensure prerelease is checked.
3. Publish.

24 hours after CI has completed and artifacts are created:
1. Uncheck prerelease, and save.
2. Update channel server

> The resulting CICD run can be viewed here: 

Reviewer 2: ```suggestion
The resulting CI/CD run can be viewed here: 
```
